### PR TITLE
[DOCS] Combine search API and URI search API reference docs

### PR DESCRIPTION
**Pull request by jrodewig**
The search API and URI search pages document the same `_search` API.
This combines the documentation from each page under the search API
docs.

Changes:

* Adds an abbreviated title for the search API page.
* Removes the following invalid query parameters:
  * `analyzer`
  * `analyze_wildcard`
  * `default_operator`
  * `df`
  * `lenient`
  * `suggest_mode`
  * `suggest_size`
* Removes the URI search docs page and adds a related redirect.
* Updates the headings of several examples

~I'll treat the redirect as a breaking change and only apply to the `master` branch. I'll open up separate PRs to remove the invalid query parms on maintained branches.~ We're now backporting the redirect to all maintained branches.

Relates to #48194

### CODE REVIEW

File: docs/reference/search/search.asciidoc [hunk 1]
```diff
@@ -1,65 +1,56 @@
...
-`analyzer`::
-  (Optional, string) Defines the analyzer to use for the query string.
-
-`analyze_wildcard`::
-  (Optional, boolean) If `true`, wildcard and prefix queries will also be
-  analyzed. Defaults to `false`.
+(Optional, boolean)
+If `true`, partial results are returned for request timeouts one or
+<<shard-failures,partial shard failures>>. If `false`, an error is returned with
+no partial results. Defaults to `true`.
```
Comment (debadair) [inline docs/reference/search/search.asciidoc:50]:
```suggestion
If `true`, returns partial results if there are request timeouts or
<<shard-failures,shard failures>>. If `false`, returns an error with
no partial results. Defaults to `true`.
```

File: docs/reference/search/search.asciidoc [hunk 2]
```diff
@@ -1,65 +1,56 @@
...
++
+You can override the default for this field by setting
+`search.default_allow_partial_results` to `false` in your cluster settings.
```
Comment (debadair) [inline docs/reference/search/search.asciidoc:53]:
```suggestion
To override the default for this field, set the 
`search.default_allow_partial_results` cluster setting to `false`.
```

**Review by debadair — CHANGES_REQUESTED** [review]
Left a couple suggested edits--the one sentence definitely needs tweaked one way or another. :-)

File: docs/reference/search/search.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/search/search.asciidoc]: Thanks for catching this one.

**Comment by jrodewig** [comment]
Thanks for catching those @debadair. I've committed your suggestions.

**Review by debadair — APPROVED** [review]
LGTM

**Comment by jrodewig** [comment]
### Backport commits
master e6542c0
7.x a528319 
7.7 804c38f 
7.6  ac36232

**Pull request merged**